packages/dashboard-frontend/src/services/registry/devfiles.ts

```diff
@@ -71,6 +116,44 @@ export async function fetchRegistryMetadata(registryUrl: string): Promise<che.De
   }
 }
 
+function getExternalRegistryMetadataFromStorage(url: string): che.DevfileMetaData[] | undefined {
+  const externalRegistries = SessionStorageService.get(SessionStorageKey.EXTERNAL_REGISTRIES);
+  if (typeof externalRegistries !== 'string') {
+    return undefined;
+  }
+  try {
+    const externalRegistriesObj = JSON.parse(externalRegistries);
+    return externalRegistriesObj[url]?.metadata;
+  } catch (e) {
+    return undefined;
+  }
+}
+
+function setExternalRegistryMetadataToStorage(url: string, metadata: che.DevfileMetaData[]): void {
```

Comment:
Can you elaborate on why we need to keep external registries metadata in the SessionStorage?

Reply:
Do you happen to know how many requests are typically sent to devfile.io to obtain the list of devfiles and their metadata?

Reply:
I don't know how many requests are typically sent to devfile.io. 

I got this requirement from the additional context of the original issue: https://github.com/eclipse/che/issues/22142 :

**"For performance optimization, we should read, https://registry.devfile.io/index only once during UD pod startup.
Also, it would be nice to add the devfile.io tag on UD to all devfiles coming from https://registry.devfile.io..."**

So, we shouldn't send a new request, each time when we resolve any factory.